Source/WarpX.H

```diff
@@ -506,6 +521,9 @@ public:
     [[nodiscard]] const amrex::MultiFab&
     getField(warpx::fields::FieldType field_type, int lev, int direction = 0) const;
 
+    [[nodiscard]] const amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>,3>>& getEfield_fp_vec() const {return Efield_fp;}
+    [[nodiscard]] const amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>,3>>& getBfield_fp_vec() const {return Bfield_fp;}
```

Comment:
The only alternative I see is to make the ImplicitSolver class a friend class of WarpX. Dave G. and I discussed this briefly and concluded that accessors are the better way.

Accessors are pretty common. Any particular reason why you are against it?

Reply:
It is still using accessors, just that the way it is called has changed. These routines could be changed to use the `FieldType` enumerator to specify which field to return.

Reply:
Yes, agreed. I can update the PR accordingly.

Reply:
clang tidy complained about not having. return type for the default switch case in getMulitLevelField.

I added one below the ABORT message so clang tidy would be happy, but this may not be what you want. Take a look.